### PR TITLE
Working copy support

### PR DESCRIPTION
**Pull request by sneridagh**

### CODE REVIEW

File: docs/source/workingcopy.rst [hunk 1]
```diff
@@ -0,0 +1,228 @@
+Working Copy
+============
+
+Plone has the "Working copy" feature provided by the core package ``plone.app.iterate``.
+It allows the user to create a working copy of a (published or live) content object and
```
Comment (tisto) [inline docs/source/workingcopy.rst:5]: "the user" -> "users" (minor)

File: docs/source/workingcopy.rst [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+work with it until it's ready to be published without having to edit the original object.
+
+This process has several steps of it's life cycle:
+
+Check-out
```
Comment (tisto) [inline docs/source/workingcopy.rst:10]: "Create working Copy (aka Check-out)"

File: docs/source/workingcopy.rst [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "@id": "http://localhost:55001/plone/working_copy_of_document",
+}
+
+the GET content of any object, also states the location of the working copy, if any (``working_copy``).
+
+GET /plone/document HTTP/1.1
```
Comment (tisto) [inline docs/source/workingcopy.rst:45]: I guess in the future we want to replace this with an auto-generated example (which is not possible right now of course).

File: docs/source/workingcopy.rst [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+  }
+}
+
+Check-in
+---------
+
+Once the user has finished editing the working copy and wants to update the original
+with the changes in there, or "check-in" the working copy.
+
+PATCH /plone/document/@workingcopy HTTP/1.1
```
Comment (tisto) [inline docs/source/workingcopy.rst:203]: @sneridagh I am unsure about that one. Though, I do not have any obvious better choice. It feels a bit odd to do a PATCH on the working copy when you actually replace (and change) the original document instead. I guess we have to live with that double-meaning of "@workingcopy" that for GET/POST it represents the working copy object and for PATCH it is a working-copy operation. I need to think about this and maybe sleep over it. I am just talking out loud here from the top of my head.

**Comment by tisto** [comment]
@sneridagh overall this looks good to me! I just mentioned a few typos / small improvements and I would like to think about the PATCH operation, then we are good to go I guess. :)

**Review by tisto — APPROVED** [review]
LGTM!

File: docs/source/workingcopy.rst (hunk 4 above)
Comment (sneridagh) [reply docs/source/workingcopy.rst:203]: The PATCH here is in the original document, so it's the one to be changed. Well, let's see how it goes the implementation.

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@sneridagh the only thing missing is a changelog entry :)

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by sneridagh** [comment]
@tisto I think it's ready. 
What I would do: Merge this as it is, then create a new one removing all the working copy Plone 4 related amendments. So, if someone wants to backport it like this to 7.x still will be able to do it, cherry picking this PR. What do you think?

File: .github/workflows/zpretty.yml [hunk 5]
```diff
@@ -34,6 +34,7 @@ jobs:
       - name: run zpretty
         run: find src/ -name *.zcml | xargs zpretty -i
 
-      # run git diff
-      - name: run git diff
-        run: git diff --exit-code
+      # XXX: this doesn't work on gh actions (https://github.com/plone/plone.restapi/pull/1119/checks?check_run_id=2686474411)
```
Comment (tisto) [inline .github/workflows/zpretty.yml:37]: @tiberiuichim zpretty failed on this PR after I merged master. Even after I ran zpretty locally if failed on gh actions. No clue why...

**Comment by tisto** [comment]
@sneridagh we need to freeze the timestamp of the working copy. Otherwise, our docs diff complains.

File: .github/workflows/zpretty.yml (hunk 5 above)
Comment (tiberiuichim) [reply .github/workflows/zpretty.yml:37]: I'd say it produces either different whitespace or line endings.

**Comment by sneridagh** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@tiberiuichim @terapyon do you have time for a review of this PR today? I'd like to merge and release this today or tomorrow if possible and a third or fourth review would be nice to have.

**Pull request merged**